go/tools/gopackagesdriver/archive_to_json.go

```diff
@@ -0,0 +1,233 @@
+	}
+	return nil
+}
+
+func (p *bazelArgsParser) Parse(arguments []string) error {
+	if len(arguments) == 0 {
+		return nil
+	}
+	firstArgument := arguments[0]
+
+	// If the first argument is not a param argument,
+	// read the tokens from the arguments.
+	if firstArgument[0] != '@' {
+		scanner := newSliceScanner(arguments)
+		return p.parseBazelArguments(scanner)
+	}
+	if len(arguments) > 1 {
+		return errors.New("expected single argument with param file")
+	}
+
+	// We have a param argument, read tokens from the paramsReader
+	// using the param argument value.
+	paramsReader, err := p.paramsReaderGetter(firstArgument[1:])
+	if err != nil {
+		return err
+	}
+	scanner := bufio.NewScanner(paramsReader)
+	return p.parseBazelArguments(scanner)
+}
+
+func NewBazelArgsParser() *bazelArgsParser {
+	return &bazelArgsParser{
+		args: map[string]arg{},
+		paramsReaderGetter: func(value string) (io.Reader, error) {
+			return os.Open(value)
+		},
+	}
+}
+
+type archive struct {
+	ID              string            `json:"ID"`
+	PkgPath         string            `json:"PkgPath"`
+	ExportFile      string            `json:"ExportFile"`
+	GoFiles         []string          `json:"GoFiles"`
+	CompiledGoFiles []string          `json:"CompiledGoFiles"`
+	OtherFiles      []string          `json:"OtherFiles"`
+	Imports         map[string]string `json:"Imports"`
+}
+
+// Parse a delimiter escaped string, for instace:
+// value = "foo\\\==\\ba\=r, delimiter = '=', escape = '\'
+// returns ["foo\=", "\ba=r"]
+func splitEscapedString(value string, delimiter, escape rune) []string {
+	var field strings.Builder
+	escaped := false
+	values := []string{}
+	for _, r := range value {
+		switch {
+		case escaped:
+			// The previous character was the escape char,
+			// so we treat the current character as literal.
+			field.WriteRune(r)
+			escaped = false
+		case r == escape:
+			// We see an escape character, so we mark that
+			// the next character should be taken literally.
+			escaped = true
+		case r == delimiter:
+			// We reached a real delimiter (not escaped).
+			// Finish the current field and start a new one.
+			values = append(values, field.String())
+			field.Reset()
+		default:
+			// Normal character, just append.
+			field.WriteRune(r)
+		}
+	}
+	// Add the last field if any
+	return append(values, field.String())
+}
+
+func parseKeyValueArg(value string, m map[string]string) error {
+	fields := splitEscapedString(value, '=', '\\')
+	if len(fields) != 2 {
+		return fmt.Errorf("could not parse key value pair: %s", value)
+	}
+	m[fields[0]] = fields[1]
+	return nil
+}
+
+// The bazel args will be parsed either from the program args or
+// from a file if its path prepended with "@" is provided.
+func parseArchiveAndOutputPath(arguments []string, archive *archive) (string, error) {
+	var outputPath string
+	parser := NewBazelArgsParser().
+		WithArg("--id", func(value string) {
+			archive.ID = value
+		}).
+		WithArg("--pkg-path", func(value string) {
```

Comment:
Let's use the `flag` package for argument parsing. I'd also suggest using `_` instead of `-` for consistency with other commands.